File: Lib/FPVR/Entities/VehicleInfo.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace FPVR
{
    /// <summary>
    /// Defines the different types of vehicle.
    /// </summary>
    public enum VehicleType
    {
        Unknown,
        Drone,
        Plane,
        Helicopter,
        Car,
        Truck,
        Boat,
        Hovercraft,
        Other
    };
    
    /// <summary>
    /// Provides information about a specific vehicle.
    /// </summary>
    public class VehicleInfo
    {
        /// <summary>
        /// Gets or sets an ID for the vehicle that is unique to the provider.
        /// </summary>
        public string Id { get; set; }

        /// <summary>
        /// Gets or sets the 3D model name, if any, that can be used to represent the vehicle.
        /// </summary>
        public string ModelName { get; set; }

        /// <summary>
        /// Gets or sets the name of the vehicle. 
        /// </summary>
        public string Name { get; set; }

        /// <summary>
        /// Gets or sets the provider that contains the vehicle.
        /// </summary>
        public IVehicleProvider Provider { get; set; }

        /// <summary>
        /// Gets or set the general type of the vehicle.
        /// </summary>
        public VehicleType Type { get; set; }
    }
}

File: Android/FPVR/app/src/main/java/com/solersoft/fpvr/fpvrlib/VehicleInfo.java
package com.solersoft.fpvr.fpvrlib;

/**
 * Created by jbienz on 5/20/2015.
 */
public class VehicleInfo
{
    //region Member Variables
    private String id;
    private String modelName;
    private String name;
    private IVehicleProvider provider;
    private VehicleType type = VehicleType.Unknown;
    //endregion

    /**
     * Gets an ID for the vehicle that is unique to the provider.
     * @return An ID for the vehicle that is unique to the provider.
     */
    public String getId(){ return id; }

    /**
     * Sets an ID for the vehicle that is unique to the provider.
     * @param value An ID for the vehicle that is unique to the provider.
     */
    public void setId(String value) { id = value; }

    /**
     * Gets the 3D model name, if any, that can be used to represent the vehicle.
     * @return The 3D model name, if any, that can be used to represent the vehicle.
     */
    public String getModelName(){ return modelName; }

    /**
     * Sets the 3D model name, if any, that can be used to represent the vehicle.
     * @param value The 3D model name, if any, that can be used to represent the vehicle.
     */
    public void setModelName(String value) { modelName = value; }

    /**
     * Gets the name of the vehicle.
     * @return The name of the vehicle.
     */
    public String getName(){ return name; }

    /**
     * Sets the name of the vehicle.
     * @param value The name of the vehicle.
     */
    public void setName(String value) { name = value; }

    /**
     * Gets the provider that contains the vehicle.
     * @return The provider that contains the vehicle.
     */
    public IVehicleProvider getProvider(){ return provider; }

    /**
     * Sets the provider that contains the vehicle.
     * @param value The provider that contains the vehicle.
     */
    public void setProvider(IVehicleProvider value) { provider = value; }

    /**
     * Gets the general type of the vehicle.
     * @return The general type of the vehicle.
     */
    public VehicleType getType(){ return type; }

    /**
     * Sets the general type of the vehicle.
     * @param value The general type of the vehicle.
     */
    public void setType(VehicleType value) { type = value; }
}
